bundles/persistence/org.openhab.persistence.jpa/java/org/openhab/persistence/jpa/internal/JpaPersistenceService.java

```diff
@@ -211,8 +211,12 @@ protected EntityManagerFactory newEntityManagerFactory() {
 		Map<String, String> properties = new HashMap<String, String>();
 		properties.put("javax.persistence.jdbc.url", JpaConfiguration.dbConnectionUrl);
 		properties.put("javax.persistence.jdbc.driver", JpaConfiguration.dbDriverClass);
-		properties.put("javax.persistence.jdbc.user", JpaConfiguration.dbUserName);
-		properties.put("javax.persistence.jdbc.password", JpaConfiguration.dbPassword);
+		if(JpaConfiguration.dbUserName != null) {
+		    properties.put("javax.persistence.jdbc.user", JpaConfiguration.dbUserName);
+		}
+		if(JpaConfiguration.dbPassword != null) {
+		    properties.put("javax.persistence.jdbc.password", JpaConfiguration.dbPassword);
+		}
```

Comment:
Ok - although this might also confuse since my application is where no user/pass is required since the database is actually embedded - it’s not a server…

How about if user is not null and password is null, then add a warning? I’m guessing that any server based system will at least require a login name.